google/cloud/forseti/services/scanner/dao.py

```diff
@@ -150,14 +254,21 @@ def list(self, inventory_index_id=None):
                 list: List of Violation row entry objects.
             """
             with self.violationmaker() as session:
-                if inventory_index_id:
+                scanner_index = (
+                    session.query(ScannerIndex)
+                    .order_by(ScannerIndex.created_at_datetime.desc()).first())
```

Comment:
A few comments here:
* This `list()` should also support `scanner_index_id` as a parameter.
* If only `scanner_index_id` is given, then find the violations for it.
* If only `inventory_index_id` is given, then find the `scanner_index_id` with the last `SUCCESS` from the `ScannerIndex` table.  Then use the found `scanner_index_id` to find the violations.
* If neither are given, then find the last `scanner_index_id`.
* If both are given... how do you think this should be handled?

Reply:
Here's what I did:
 * if both indices are supplied we fillter on `inventory_index_id` *and* `scanner_index_id`
 * if neither index is supplied we return all violations (i.e. no filtering whatsoever)
 * if either index is supplied we filter on that index

That seemed the most logical to me.